src/Microsoft.ML.Core/Utilities/OrderedWaiter.cs

```diff
@@ -90,7 +90,7 @@ public void Wait(long position, CancellationToken token = default(CancellationTo
                 ev = new WaitStats(position);
                 _waiters.Add(ev);
             }
-            ev.Event.Wait(token);
+            ev.Event.Wait(5*60*1000, token);
             if (_ex != null)
```

Comment:
This is a five minute wait. Why five minutes? Can you please give more background behind this choice?

Reply:
This caused most of our test hanging according to my local test result.
Some processor is waiting for signal to start but seems in some case it never can get the signal to run and the wait hangs the test process. This happens in multiple test cases so I just putting a pretty large timeout here, maybe we need discuss more to understand the root cause

---
In reply to: [350531489](https://github.com/dotnet/machinelearning/pull/4503#discussion_r350531489) [](ancestors = 350531489)